lib/db_connection/connection.ex

```diff
@@ -593,11 +605,15 @@ defmodule DBConnection.Connection do
   defp start_timer(pid, timeout) do
     :erlang.start_timer(timeout, self(), {__MODULE__, pid, timeout})
   end
+  defp start_timer(_, :infinity, _), do: nil
+  defp start_timer(pid, timeout, msg) do
+    :erlang.start_timer(timeout, pid, msg)
+  end
 
   defp cancel_timer(nil), do: :ok
   defp cancel_timer(timer) do
     case :erlang.cancel_timer(timer) do
-      false -> flush_timer(timer)
+      false -> :erlang.read_timer(timer) and flush_timer(timer)
```

Comment:
`read_timer` will always be `false` here so wont flush, we don't need this part.